indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -835,26 +844,24 @@ void syncFromStorage()
 
     try {
       if (active) {
-        final Map<String, Task> newTasks =
-            CollectionUtils.toMap(taskStorage.getActiveTasks(), Task::getId, Function.identity());
-        final Map<String, Task> oldTasks =
-            CollectionUtils.mapValues(activeTasks, entry -> entry.task);
+        final Map<String, TaskInfo<Task, TaskStatus>> newTasks = CollectionUtils.toMap(taskStorage.getActiveTaskInfos(), (taskInfo) -> taskInfo.getTask().getId(), Function.identity());
+        final Map<String, TaskInfo<Task, TaskStatus>> oldTasks = CollectionUtils.mapValues(activeTasks, entry -> entry.taskInfo);
 
         // Identify the tasks that have been added or removed from the storage
-        final MapDifference<String, Task> mapDifference = Maps.difference(oldTasks, newTasks);
-        final Collection<Task> addedTasks = mapDifference.entriesOnlyOnRight().values();
-        final Collection<Task> removedTasks = mapDifference.entriesOnlyOnLeft().values();
+        final MapDifference<String, TaskInfo<Task, TaskStatus>> mapDifference = Maps.difference(oldTasks, newTasks);
+        final Collection<TaskInfo<Task, TaskStatus>> addedTasks = mapDifference.entriesOnlyOnRight().values();
+        final Collection<TaskInfo<Task, TaskStatus>> removedTasks = mapDifference.entriesOnlyOnLeft().values();
 
         // Remove tasks not present in metadata store if their lastUpdatedTime is before syncStartTime
         int numTasksRemoved = 0;
-        for (Task task : removedTasks) {
+        for (TaskInfo<Task, TaskStatus> task : removedTasks) {
```

Comment:
Looking at this patch, I wonder if `TaskInfo<Task, TaskStatus>` shouldn't be a separate class of its own.
Probably named `TaskWithStatus`.

(I am not even sure why this class was generic in the first place. The only other possible type I see the generic
param taking is `TaskIdentifier`, which could have been its own class `TaskInfoWithId` or something. But we can address this later).